engine/grid.py

```diff
@@ -96,16 +109,170 @@ def generate_nodes(start_lat, start_long, rows, cols, step_size_m):
         self.long_min = long_min
         self.long_max = long_max
 
-        self.num_rows, self.num_cols = calc_step(lat_min, lat_max, long_min, long_max, STEP_SIZE_METERS)
+        self.num_rows, self.num_cols = calc_step(
+            lat_min, lat_max, long_min, long_max, STEP_SIZE_METERS)
+
+        self.nodes = generate_nodes(
+            lat_min, long_min, self.num_rows, self.num_cols, STEP_SIZE_METERS)
+        self.border_nodes = None
+        self.leftmost_node = None
+        self.leftmost_node_pos = None
+        self.active_nodes_list = []
```

Comment:
Can you add documentation about what active_nodes_list and inactive_nodes_list are? And what their purpose is?

Reply:
@Shreya1233